Sources/Unbox.swift

```diff
@@ -910,6 +924,24 @@ private struct UnboxArrayContainer<T: Unboxable>: UnboxableWithContext {
     }
 }
 
+private struct UnboxWithContextContainer<T: UnboxableWithContext>: UnboxableWithContext {
+    let model: T
+
+    init(unboxer: Unboxer, context: UnboxContainerContext) {
+        let modelContext: T.ContextType = context.context as! T.ContextType // This is safe as the context is type checked before being put into UnboxContainerContext.context
+        self.model = unboxer.unbox(context.key, isKeyPath: context.isKeyPath, context: modelContext)
+    }
+}
+
+private struct UnboxArrayWithContextContainer<T: UnboxableWithContext>: UnboxableWithContext {
+    let models: [T]
+
+    init(unboxer: Unboxer, context: UnboxContainerContext) {
+        let modelContext: T.ContextType = context.context as! T.ContextType // This is safe as the context is type checked before being put into UnboxContainerContext.context
```

Comment:
Instead of this, could you make `UnboxContainerContext` generic and have it be `Any?` for the normal usecase, and `T.ContextType` for this? That way you'd avoid this force casting (just thinking out loud here 🤔)